aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,709 @@
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfigTags(rName, image),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.foo", "bar"),
+				),
+			},
+			{
+				Config: testAccSagemakerModelConfigTagsUpdate(rName, image),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.bar", "baz"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_model.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_primaryContainerModelDataUrl(t *testing.T) {
+	rName := acctest.RandString(10)
+	modelDataUrl := fmt.Sprintf("https://s3-us-west-2.amazonaws.com/terraform-testacc-sagemaker-model-data-bucket-%s/model.tar.gz", rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerPrimaryContainerModelDataUrlConfig(rName, image, modelDataUrl),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo",
+						"primary_container.0.model_data_url", modelDataUrl),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_model.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_primaryContainerHostname(t *testing.T) {
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerPrimaryContainerHostnameConfig(rName, image),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo",
+						"primary_container.0.container_hostname", "foo"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_model.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_primaryContainerEnvironment(t *testing.T) {
+	rName := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
```

Comment:
Likely well after this pull request was introduced, we switched the majority of the codebase acceptance testing from `resource.Test()` to `resource.ParallelTest()`